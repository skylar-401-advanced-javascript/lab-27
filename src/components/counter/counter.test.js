/* globals shallow, mount, render */
import React from 'react';
import renderer from 'react-test-renderer';
import Counter from './counter';

describe('Counter Component', () => {
    it('matches snapshot without props', () => {
      const tree = renderer.create(<Counter />).toJSON()
      expect(tree).toMatchSnapshot();
    });

    it('can shallowly render without props', () => {
      let counter = shallow(<Counter />);
      expect(counter.find('section').exists()).toBe(true);
      expect(counter.find('span').exists()).toBe(true);
      expect(counter.find('a').exists()).toBe(true);
    });

    it('can simulate button up and down presses', () => {
      let counter = mount(<Counter />);
      let buttonUp = counter.find('.up');
      let buttonDown = counter.find('.down');
      expect(counter.state('count')).toBe(0);
      buttonUp.simulate('click')
      expect(counter.state('count')).toBe(1);
      buttonUp.simulate('click')
      expect(counter.state('count')).toBe(2);
      buttonUp.simulate('click')
      expect(counter.state('count')).toBe(3);
      buttonDown.simulate('click')
      expect(counter.state('count')).toBe(2);
      buttonDown.simulate('click')
      expect(counter.state('count')).toBe(1);
      buttonDown.simulate('click')
      expect(counter.state('count')).toBe(0);
    });

});
