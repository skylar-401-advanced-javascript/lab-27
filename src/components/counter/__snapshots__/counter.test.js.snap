// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component matches snapshot without props 1`] = `
<section
  className="counter"
>
  <a
    className="down clicker"
    href="#"
    onClick={[Function]}
  >
    -
  </a>
  <span
    className="count "
  >
    0
  </span>
  <a
    className="up clicker"
    href="#"
    onClick={[Function]}
  >
    +
  </a>
</section>
`;
